### ISSUE TITLE
how should i detect 2a?

### ISSUE THREAD

**Issue by bypassiwastaken** [bypassiwastaken/skiovox-helper#20]
i guess the main difference is that new windows persist. does anyone know an elegant way to do this?

**Comment by lukasexists** [comment]
the easiest way would probably be just to ask the user and store it somewhere. its not like they're going to lie or forget right after doing it right?

**Comment by bypassiwastaken** [comment]
i guess, but its a bit ugly. also someone could do 2a and 2b so its not necessarily gonna be true forever. there must be an automatic way. ill test later

**Comment by zek-c** [comment]
theres certain things that only 2a users can do so i guess if you test something and if it doesnt return error assume they are on 2A

**Comment by GoodMusic8596** [comment]
Perhaps have a settings menu perhaps and have the user select and let them know what the benefits/downsides are

**Comment by BadPingHere** [comment]
My ideas right now is:
1. If you open a full screen popup, once you open the new tab, it will remove focus from the new tab, which can be detectable. However I am not sure if this is fullproof, since I don't have a 2A device to check this on. Moreover, this is quite disruptive, even if you remove it in a split second.

2. If you check every time, or every so often, how many tabs are open on the new tab, and then open a 1x1, really tiny webpage popup on the new tab, for 2A users the popup should last, while 2B/2C it shouldnt even register. This should be easy to make, though again I don't have a clue on how to replicate 2A without a proper chromebook for it.

**Comment by zek-c** [comment]
problem with that, the window can't go small. the minimum size is about half the screen.

**Comment by BadPingHere** [comment]
> problem with that, the window can't go small. the minimum size is about half the screen.

yeah, you are right. It seems the minimum size is around w=160 h=100, but this still creates a small popup. Make its color the same as the background, and only do it every so often, it can work.

**Comment by nicolasog13** [comment]
when you first open, a button/banner could be on the newtab page that asks you to fill out a form. it would be a separate HTML file on the extension that asks if you are 2a or 2b/c, and you click the save button (not sure how efficient that would be for coding tho)

**Comment by 3kh0** [comment]
Just ask the user 🙃

**Comment by nicolasog13** [comment]
you either have to find a way to do this automatically or just ask the user :P

**Comment by zek-c** [comment]
well obviously are you dumb

**Comment by Su4o** [comment]
You can detect 2A by trying to make the user request to try to, CTRL + N. 2A members can open multiple windows, but 2B and 2C can't. So if a new window does open, you can now know that it's 2A. I don't know if this would work tho. The best option is just to tell the user. Why would they lie about if they are a 2A member, 2B member, 2C member? If they "accidentally" choose a different option, you should make an option for them to change.

**Comment by zek-c** [comment]
> You can detect 2A by trying to make the user request to try to, CTRL + N. 2A members can open multiple windows, but 2B and 2C can't. So if a new window does open, you can now know that it's 2A. I don't know if this would work tho. The best option is just to tell the user. Why would they lie about if they are a 2A member, 2B member, 2C member? If they "accidentally" choose a different option, you should make an option for them to change.

not true, im on 2C and can open multiple windows IF the new window has a request if you really want to close it when you try to close it. would be a nice feature to create a window that has the confirm close to make it easier for 2b and 2c

**Comment by Su4o** [comment]
@zek-c then, it's just easy to just give the user the choice to tell with the method of the exploit there doing :p

**Comment by zek-c** [comment]
> i guess the main difference is that new windows persist. does anyone know an elegant way to do this?

when extension is downloaded , open a window that takes around 5 seconds to load, do window.onload() to set a localstorage value, if the window doesnt load because  2c and 2b will close the window instantly, the value wont be set
